pallets/parachain-system/src/benchmarking.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Benchmarking setup for pallet-collator-selection
+
+#![allow(unused)]
+
+use super::*;
+
+use crate::{Pallet, PendingUpwardMessages, PendingValidationCode, ValidationData};
+use cumulus_primitives_core::relay_chain::v2::HeadData;
+use frame_benchmarking::{account, benchmarks, impl_benchmark_test_suite, whitelisted_caller};
+use frame_support::{
+	assert_ok,
+	codec::Decode,
+	traits::{Currency, EnsureOrigin, Get},
+};
+use frame_system::{EnsureNone, EnsureRoot, EnsureSigned, EventRecord, RawOrigin};
+use sp_core::H256;
+use sp_std::{collections::btree_set::BTreeSet, prelude::*};
+use sp_trie::StorageProof;
+
+const SEED: u32 = 0;
+
+benchmarks! {
+	set_validation_data_no_messages {
+		// root and proof are obtained via `RelayStateSproofBuilder::default().into_state_root_and_proof()`
```

Comment:
This is the same problem I had here https://github.com/paritytech/cumulus/pull/1709  
Do you plan on making the mocked proof builder no-std compatible?

Reply:
I looked into it, but unfortunately seemed somewhat of an effort with my limited understanding and some dependent crates themselves being no-std :(

Reply:
Or would you maybe have an estimate how long it would take? Maybe I can try anyway and ask for help if I'm stuck somewhere.

Reply:
The base version already works fine. But updating all to no-std was annoying. I can take another look at it though.  
Currently there is no timeline. If it does not work we may merge the solution with hard-coded proof blobs.